packetery/src/Packetery/Carrier/OptionsPage.php

```diff
@@ -0,0 +1,351 @@
+		$form     = $this->formFactory->create( $optionId );
+		$form->setAction( $this->httpRequest->getUrl() );
+
+		$container = $form->addContainer( $optionId );
+
+		$container->addCheckbox(
+			'active',
+			__( 'Active carrier', 'packetery' )
+		);
+
+		$container->addText( 'name', __( 'Display name', 'packetery' ) )
+					->setRequired()
+					->addRule( $form::MIN_LENGTH, __( 'Carrier display name must have at least 2 characters!', 'packetery' ), 2 );
+
+		$weightLimits = $container->addContainer( 'weight_limits' );
+		if ( empty( $carrierData['weight_limits'] ) ) {
+			$this->addWeightLimit( $weightLimits, 0 );
+		} else {
+			foreach ( $carrierData['weight_limits'] as $index => $limit ) {
+				$this->addWeightLimit( $weightLimits, $index );
+			}
+		}
+
+		$surchargeLimits = $container->addContainer( 'surcharge_limits' );
+		if ( empty( $carrierData['surcharge_limits'] ) ) {
+			$this->addSurchargeLimit( $surchargeLimits, 0 );
+		} else {
+			foreach ( $carrierData['surcharge_limits'] as $index => $limit ) {
+				$this->addSurchargeLimit( $surchargeLimits, $index );
+			}
+		}
+
+		$item = $container->addText( 'free_shipping_limit', __( 'Free shipping limit', 'packetery' ) );
+		$item->addRule( $form::FLOAT, __( 'Please enter a valid decimal number.', 'packetery' ) );
+		$container->addHidden( 'id' );
+
+		$carrierOptions       = get_option( $optionId );
+		$carrierOptions['id'] = $carrierData['id'];
+		if ( empty( $carrierOptions['name'] ) ) {
+			$carrierOptions['name'] = $carrierData['name'];
+		}
+		$container->setDefaults( $carrierOptions );
+
+		return $form;
+	}
+
+	/**
+	 * Validates options.
+	 *
+	 * @param array $options Packetery_options.
+	 *
+	 * @return array
+	 */
+	public function validateOptions( array $options ): array {
+		if ( ! empty( $options['id'] ) ) {
+			$options = $this->validateAndMergeNewLimits( $options, 'weight_limits' );
+			$options = $this->checkOverlappingAndSort(
+				$options,
+				'weight_limits',
+				'weight',
+				__( 'Weight rules are overlapping, fix it please.', 'packetery' )
+			);
+			$options = $this->validateAndMergeNewLimits( $options, 'surcharge_limits' );
+			$options = $this->checkOverlappingAndSort(
+				$options,
+				'surcharge_limits',
+				'order_price',
+				__( 'Surcharge rules are overlapping, fix it please.', 'packetery' )
+			);
+
+			$form     = $this->createForm( $options );
+			$optionId = 'packetery_carrier_' . $options['id'];
+			$form[ $optionId ]->setValues( $options );
+			if ( $form->isValid() === false ) {
+				foreach ( $form[ $optionId ]->getControls() as $control ) {
+					if ( $control->hasErrors() === false ) {
+						continue;
+					}
+					add_settings_error( $control->getCaption(), esc_attr( $control->getName() ), $control->getError() );
+					$options[ $control->getName() ] = '';
+				}
+			}
+		}
+
+		return $options;
+	}
+
+	/**
+	 * Save data if validated.
+	 */
+	public function processForm(): void {
+		$post = $this->httpRequest->getPost();
```

Comment:
Zkuste prosím použít zpracování formulářů doporučeným způsobem, tak jak to řeší nette forms (viz. dokumentace). Pokud bychom měli opravdu důvod to obcházet,  napište mi prosím proč konkrétně.